src/rust/ide/lib/parser/src/api.rs

```diff
@@ -91,17 +153,20 @@ fn to_json_single_line(val:&impl Serialize) -> std::result::Result<String,serde_
 impl<M:Metadata> ParsedSourceFile<M> {
     /// Serialize to the SourceFile structure,
     pub fn serialize(&self) -> std::result::Result<SourceFile,serde_json::Error> {
-        let code                  = self.ast.repr();
-        let id_map                = to_json_single_line(&self.ast.id_map())?;
-        let metadata              = to_json_single_line(&self.metadata)?;
-        let id_map_start          = code.len() + METADATA_TAG.len();
-        let newlines_after_id_map = 1;
-        let metadata_start        = id_map_start + id_map.len() + newlines_after_id_map;
+        let before_tag      = "\n".repeat(NEWLINES_BEFORE_TAG);
+        let before_idmap    = "\n";
+        let before_metadata = "\n";
```

Comment:
export the literal `'\n'` as constant please

Reply:
but we don't use constant elsewhere? e.g. area.rs:914

Actually "\n" as a constant wont do much, as everyone should know that "\n" is a new line. It's like creating constant `const ZERO:usize = 0`